### ISSUE TITLE
Sinclair ACs?

### ISSUE THREAD

**Issue by satcom886** [tomikaa87/gree-remote#27]
Hi, recently we've had some Sinclair AC units installed at our house.
They also support some form of control over WiFi. I'd like to integrate them into OpenHab, but I'm not nearly experienced enough to make my own binding. So I'd like to ask, would it be possible to add support for this Sinclair unit?

I can provide any info you may need (as long as it doesn't require disassembling the AC).
I already tried the python script you made but it doesn't find it (I ran `python gree.py scan -b 255.255.255.0`).

Before you dismiss this... I have a few reasons to believe your code might need very little modification to work with this brand.

# User's perspective
When enabled using the AC remote, it makes a WiFi hotspot with a random name and the password `12345678`. Connecting to it allows to use their app called [WiFi Smart](https://play.google.com/store/apps/details?id=com.wifi.smarthome). This works only as long as the user is connected to the AC's WiFi. The app has an option to make an online account and I assume that would allow the AC to connect to your home WiFi network and be controlled even over the internet (I haven't tried that because I would prefer not to create such an account.)
# How it works under the hood
I connected to the AC's WiFi using one laptop, bridged it to its Ethernet, connected another laptop and made a WiFi hotspot. I then connected my phone and could control the AC while capturing the packets exchanged between the phone and the AC on the second laptop.

**voila**
The phone constantly sends UDP packets to port 7000, address 255.255.255.0.

# Other similarities
The features are almost identical.
- [X] 5 modes in order Auto, Cool, Dry, Fan, Heat
- [X] 3 fan speed levels (low, medium, high and auto), the manual hints at more speeds in different models
- [X] X-Fan (also only available in Cool and Dry modes)
- [X] "Health"
- [X] "Sleep" function that works in exactly the same way as the Gree ACs
- [X] The display on the AC unit can be turned off
- [X] Left-right and up-down swing
- [X] "Quiet" mode
- [X] "Turbo" mode (only in Heat and Cool) - from your docs on the Gree units this works in Dry and Cool
- [X] 8°C Heating mode
- [X] Energy-saving function

**Comment by satcom886** [comment]
Just a quick update...
I just now tried to use the GreeSmart app instead and surprisingly... IT WORKS!

**Comment by satcom886** [comment]
Update 2:
I just compiled your Android app and it works too! So it seems like it might be just an issue with your python script?

**Comment by tomikaa87** [comment]
Hi,

I'm glad to hear that the android app worked fine.

I've tried the python script, it seems to be working only when you send the scan packet to the broadcast address of the current network. I've updated the script with socket flags for broadcast but it doesn't seem to work with 255.255.255.255 nor 255.255.255.0. For me it worked fine with the broadcast address of the network:

```
C:\git\github\gree-remote\PythonCLI>python3 gree.py search -b 192.168.30.255
Searching for devices using broadcast address: 192.168.30.255
Search finished, found 2 device(s)
Binding device: 192.168.30.4 (Kitchen, ID: <kitchen id>)
Bind to <kitchen id> succeeded, key = ****
Binding device: 192.168.30.5 (Bedroom, ID: <bedroom id>)
Bind to <bedroom id> succeeded, key = ****
```

If I search on 255.255.255.0 I get a socket error:
OSError: [WinError 10051] A socket operation was attempted to an unreachable network

For 255.255.255.255 it simply doesn't find any devices. Before the patch it gave an access permission error but it's solved in the patch.
